### PR TITLE
Introduced Attempt[A] as a type alias for Either[Throwable,A] and...

### LINKED ISSUE #678: Consider replacing Either[Throwable,A] with Try[A]
Paul and I had discussed this a number of months ago but never came to a resolution. Currently, we use `Either[Throwable,A]` in lots of places, along with constructs like: `try Right(a) catch { case t: Throwable => Left(t) }`. By switching to `Try`, we'd only catch non-fatal exceptions, which may or may not be considered desirable. I lean towards using `Try` (I want a `NonFatal` exception to trigger application shutdown) but I'm eager to hear other opinions.

### PR DESCRIPTION
**Pull request by mpilquist**
…changed all catches to avoid catching VirtualMachineError.

Fixes #678.

### CODE REVIEW

File: core/src/test/scala/fs2/TestUtil.scala [hunk 1]
```diff
@@ -32,7 +34,7 @@ trait TestUtil {
     catch {
       case e: InterruptedException => throw e
       case e: TimeoutException => throw e
-      case e: Throwable => ()
+      case NonFatal(e) => ()
```
Comment (pchlupacek) [inline core/src/test/scala/fs2/TestUtil.scala]: Hm, I am just thinking loudly. In strategies and executors shouldn't we just do System.exit(-1) on all fatal exceptions after stderr them? As I am aware only exception that JVM may terminate is OOM (that we perhaps shouldn't catch at all ever), all the rest will lead us to unpredictable state. Rather would kill JVM then having it run in un-predicatable state.

File: core/src/main/scala/fs2/task.scala [hunk 2]
```diff
@@ -202,14 +202,14 @@ object Task extends Instances {
...
-    Try(a.get) match {
+    Attempt(a.get) match {
       case Left(e) => Future.now(Left(e))
       case Right(f) => f
   }))
 
   /** Create a `Future` that will evaluate `a` using the given `Strategy`. */
   def apply[A](a: => A)(implicit S: Strategy): Task[A] =
-    async { cb => S(cb(Try(a))) }
+    async { cb => S(cb(Attempt(a))) }
```
Comment (pchlupacek) [inline core/src/main/scala/fs2/task.scala]: rather than having default Attempt.apply, shouldn't we have Attempt.success and Attempt.failure constructors?

Comment (mpilquist) [inline core/src/main/scala/fs2/task.scala]: I'm fine with adding `success` and `failure` but I don't think these replace `Attempt.apply` -- rather, they complement.

File: core/src/test/scala/fs2/TestUtil.scala (hunk 1 above)
Comment (mpilquist) [inline core/src/test/scala/fs2/TestUtil.scala]: Perhaps. Akka has such behavior but lets you disable it via configuration. We might want to do the same if we decide to shutdown JVM.

**Comment by mpilquist** [comment]
@pchlupacek Just addressed both of your comments. Let me know if you have any others.

**Comment by pchlupacek** [comment]
@mpilquist excellent, thanks

**Pull request merged**